### PR TITLE
os/bluestore: tune deferred_batch_ops separately for hdd and ssd

### PR DESCRIPTION
**Pull request by liewegas**
Signed-off-by: Sage Weil <sage@redhat.com>

On hdd + nvme journal with #14399 this gets me 2800 iops sequential, 400 iops random (qd 16) via
vstart (fresh rbd image, no aging, etc.).

### CODE REVIEW

File: src/os/bluestore/BlueStore.h [hunk 1]
```diff
@@ -1789,6 +1789,7 @@ class BlueStore : public ObjectStore,
   uint64_t min_alloc_size = 0; ///< minimum allocation unit (power of 2)
   size_t min_alloc_size_order = 0; ///< bits for min_alloc_size
   uint64_t prefer_deferred_size = 0; ///< size threshold for forced deferred writes
+  int deferred_batch_ops = 0; ///< deferred batch size
```
Comment (ifed01) [inline src/os/bluestore/BlueStore.h]: Shouldn't we use atomic vars for dynamically loadable settings?

**Comment by markhpc** [comment]
On HDD with NVMe WAL/DB and QD1, I'm seeing improvements even up to to 512 batched ops (though in reality the writes currently don't coalesce that well at the blk layer).  I'd be tempted to bump the default HDD batched ops up to 128 (or higher) depending on the side effects.

**Comment by liewegas** [comment]
They coalesce nicely for sequential IO but it will be a big batch of 
random IOs to throw at the disk for other workloads.  We should verify 
that it is a win for random writes too.  I suspect we'll want to have 
something that explicitly identifies sequential (coalesced) ios before 
turning the value up too high...

**Comment by markhpc** [comment]
That was my concern as well, though interestingly it didn't seem to hurt very much for random IO.  I agree though, this works best if we can identify sequential IOs.

File: src/os/bluestore/BlueStore.h (hunk 1 above)
Comment (liewegas) [inline src/os/bluestore/BlueStore.h]: In practice I think as long as the value is a word size or less we will still get an atomic update -- just not immediately.  Which should be okay.  I'm worried that using atomics might introduce a read barrier on read?

**Comment by ifed01** [comment]
rebase?

**Pull request merged**